scripts/html_linter.py

```diff
@@ -0,0 +1,473 @@
+        python_utils.PRINT('')
+        if not self.verbose_mode_enabled:
+            python_utils.PRINT('Linting HTML files.')
+        for filepath in html_files_to_lint:
+            proc_args = htmllint_cmd_args + [filepath]
+            if self.verbose_mode_enabled:
+                python_utils.PRINT('Linting %s file' % filepath)
+            with linter_utils.redirect_stdout(stdout):
+                proc = subprocess.Popen(
+                    proc_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+                encoded_linter_stdout, _ = proc.communicate()
+                linter_stdout = encoded_linter_stdout.decode(encoding='utf-8')
+                # This line splits the output of the linter and extracts digits
+                # from it. The digits are stored in a list. The second last
+                # digit in the list represents the number of errors in the file.
+                error_count = (
+                    [int(s) for s in linter_stdout.split() if s.isdigit()][-2])
+                if error_count:
+                    error_summary.append(error_count)
+                    python_utils.PRINT(linter_stdout)
+
+        with linter_utils.redirect_stdout(stdout):
+            if self.verbose_mode_enabled:
+                python_utils.PRINT('----------------------------------------')
+            for error_count in error_summary:
+                total_error_count += error_count
+            total_files_checked = len(html_files_to_lint)
+            if total_error_count:
+                python_utils.PRINT('(%s files checked, %s errors found)' % (
+                    total_files_checked, total_error_count))
+                summary_message = (
+                    '%s   HTML linting failed, '
+                    'fix the HTML files listed above.' % _MESSAGE_TYPE_FAILED)
+                summary_messages.append(summary_message)
+            else:
+                summary_message = '%s   HTML linting passed' % (
+                    _MESSAGE_TYPE_SUCCESS)
+                summary_messages.append(summary_message)
+
+            python_utils.PRINT('')
+            python_utils.PRINT(summary_message)
+            python_utils.PRINT('HTML linting finished.')
+            python_utils.PRINT('')
+
+        return summary_messages
+
+    def perform_all_lint_checks(self):
+        """Perform all the lint checks and returns the messages returned by all
+        the checks.
+
+        Returns:
+            all_messages: str. All the messages returned by the lint checks.
+        """
+        all_messages = []
+        if not self.all_filepaths:
+            python_utils.PRINT('')
+            python_utils.PRINT(
+                'There are no HTML files to lint.')
+            return all_messages
+
+        html_linter_messages = self._lint_html_files()
+
+        all_messages += html_linter_messages
+        return all_messages
+
+
+def perform_all_lint_checks(
```

Comment:
Removed.

Reply:
@seanlip Do you want this?

Reply:
First, every filetype-specific script should have a similar format regarding what is exposed to the main script.

Second -- I would say, it depends. If the main script needs to manage tasks as a whole, then the function here should give it a list of tasks and the main script can run these tasks based on the required concurrency. If it's possible for the individual linters to handle it, then it's fine to have this here and call it from the main script. But every filetype-script should be treated the same way.

@kevinlee12 is this in line with your thinking? Also @Hudda what do you think?

Reply:
@seanlip I think what I have done here is better because it provides much more encapsulation and also the code in main linting script looks much cleaner. Also, we do not have to worry about the two calls for a single type of file extension. If we do this, then we just have to call a single method and collect both 'custom' and 'third_party' linters and run those in the main linting script.

Reply:
Then I have no objection. Perhaps you could ask @kevinlee12 for the rationale behind his comment so that you have all the data to make a decision. Then make a decision, prioritizing cleanliness, understandability and maintainabiity of code, as well as some of the general principles I mentioned in the review.

Reply:
Although one thing confused me in the last sentence of your comment. Calling a single method sounds great, and having that method collect both linters sounds great. I just don't really understand what you mean by "run those in the main linting script". If you are doing that (i.e. just getting the linters and running them in the main script), then the function name here needs to be changed to get_* because it is not actually performing the lint checks.

Reply:
@seanlip will do that.

Reply:
> Although one thing confused me in the last sentence of your comment. Calling a single method sounds great, and having that method collect both linters sounds great. I just don't really understand what you mean by "run those in the main linting script". If you are doing that (i.e. just getting the linters and running them in the main script), then the function name here needs to be changed to get_* because it is not actually performing the lint checks.

@seanlip I will change the method name to `get_lint_task`.

Reply:
get_lint_tasks, if you're returning a list (plural).

Reply:
@seanlip sure.

Reply:
> First, every filetype-specific script should have a similar format regarding what is exposed to the main script.
> 
> Second -- I would say, it depends. If the main script needs to manage tasks as a whole, then the function here should give it a list of tasks and the main script can run these tasks based on the required concurrency. If it's possible for the individual linters to handle it, then it's fine to have this here and call it from the main script. But every filetype-script should be treated the same way.
> 
> @kevinlee12 is this in line with your thinking? Also @Hudda what do you think?

@kevinlee12 what do you think? Should I move 'task creation' logic to filetype scripts?

Reply:
I'd prefer for the task handling be done at the main script so that we can have more granular control, the way that was done when I lgtm-ed it is fine.